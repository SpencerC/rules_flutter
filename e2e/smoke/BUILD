"""Provides a simple way to test your rules as an external workspace.
Add a basic smoke-test target below.
"""

load("@bazel_skylib//rules:build_test.bzl", "build_test")
load("@rules_flutter//flutter:defs.bzl", "dart_library")

# Test basic rule functionality
dart_library(
    name = "test_dart_lib",
    srcs = ["test_lib.dart"],
)

build_test(
    name = "smoke_test",
    targets = [
        ":test_dart_lib",
        # Test that we can load the main rules
        "@rules_flutter//flutter:resolved_toolchain",
    ],
)

test_suite(
    name = "integration_tests",
    tests = [
        ":smoke_test",
        "//flutter_app:hello_world_app_test",
        "//flutter_app:hello_world_build_test",
        "//multiplatform:multiplatform_build_test",
    ],
)
