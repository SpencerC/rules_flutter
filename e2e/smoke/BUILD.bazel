"""Provides a simple way to test your rules as an external workspace.
Add a basic smoke-test target below.
"""

load("@bazel_gazelle//:def.bzl", "gazelle", "gazelle_binary")
load("@bazel_skylib//rules:build_test.bzl", "build_test")
load("@rules_flutter//flutter:defs.bzl", "dart_library")

gazelle_binary(
    name = "gazelle_bin",
    languages = [
        "@bazel_skylib_gazelle_plugin//bzl",
        "@rules_flutter//gazelle/flutter",
    ],
)

gazelle(
    name = "gazelle",
    gazelle = "gazelle_bin",
)

# Test basic rule functionality
dart_library(
    name = "test_dart_lib",
    srcs = ["test_lib.dart"],
)

build_test(
    name = "smoke_test",
    targets = [
        ":test_dart_lib",
        # Test that we can load the main rules
        "@rules_flutter//flutter:resolved_toolchain",
    ],
)

build_test(
    name = "single_pub_dependency_build_test",
    targets = ["@pub_intl_utils//:intl_utils"],
)

test_suite(
    name = "integration_tests",
    tests = [
        ":single_pub_dependency_build_test",
        ":smoke_test",
        "//dart_package:dart_package_test",
        "//flutter_app:app_build_test",
        "//multiplatform:multiplatform_build_test",
    ],
)

alias(
    name = "go",
    actual = "@rules_go//go",
)
