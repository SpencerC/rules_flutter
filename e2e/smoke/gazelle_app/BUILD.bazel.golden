load("@rules_flutter//flutter:defs.bzl", "flutter_library")

flutter_library(
    name = "lib",
    srcs = ["lib/main.dart"],
    pubspec = "pubspec.yaml",
    deps = [
        "@flutter_sdk//flutter/packages/flutter",
        "@flutter_sdk//flutter/packages/flutter_test",
        "@pub_cupertino_icons//:cupertino_icons",
    ],
)
